### DISCUSSION TITLE
Jint version used by Salesforce Marketing Cloud returns wrong Bitwise left result

### DISCUSSION THREAD

**Discussion by attilah** [sebastienros/jint#1401] [Q&A]
Tried to implement hmac sha-256 for marketing cloud in SSJS which is powered by some old version of Jint and it turns out that most of the bitwise operations are either returning wrong result or throwing ArithmeticOverflowException.

Simplest issue I ran into is this:
```
1359893119 << 26

Returns: 9.12608823775068e+16
```

Correct value would be: ```-67108864```

Do you recall any issue around this Sebastien?

This shows the issue in more detail: https://salesforce.stackexchange.com/questions/363226/ssjs-left-shift-operator-and-2s-complement-different-behaviour-than-in-j

**Comment by lahma** [comment]
Latest Jint 3 beta seems to work correctly:

```
jint> 1359893119 << 26
-67108864
```

So the library should be updated or some custom JS function to be used (can also be CLR function if this SSJS thing allows registering those).

**Comment by attilah** [comment]
@lahma unfortunately CLR is not enabled (for obvious security reasons) in SFMC :-( I doubt that Salesforce will update the JS runtime engine if they haven't done it in years :-( stuck at ES3...I think it says it all.

Trying to get a workaround with custom JS but it is far from ideal.

**Comment by lahma** [comment]
I don't know if Salesforce has any issue tracker to file an issue about updating. They would benefit in many ways, like new syntax support, manyfold performance increase in some scenarios.

**Reply by attilah** [reply]
They have idea hub, we'll try to ask about it on other channels.

**Reply by lahma** [reply]
Great, it would be nice if you ever hear back from them to update us too. Adoption for latest Jint is always good 🙂

**Comment by attilah** [comment]
I added my feedback on the closest thing as an issue tracker they've for the public: https://ideas.salesforce.com/s/idea/a0B8W00000GdfBLUAZ/update-ssjs-to-use-modern-javascript
